api4/channel.go

```diff
@@ -2094,3 +2097,87 @@ func moveChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		mlog.Warn("Error while writing response", mlog.Err(err))
 	}
 }
+
+func getTopChannelsForTeamSince(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId().RequireTimeRange()
+	if c.Err != nil {
+		return
+	}
+
+	team, err := c.App.GetTeam(c.Params.TeamId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), team.Id, model.PermissionViewTeam) {
+		c.SetPermissionError(model.PermissionViewTeam)
+		return
+	}
+
+	topChannels, err := c.App.GetTopChannelsForTeamSince(c.Params.TeamId, c.AppContext.Session().UserId, &model.InsightsOpts{
+		StartUnixMilli: c.Params.TimeRange,
+		Page:           c.Params.Page,
+		PerPage:        c.Params.PerPage,
+	})
+
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	js, jsonErr := json.Marshal(topChannels)
+	if jsonErr != nil {
+		c.Err = model.NewAppError("getTopChannelsForTeamSince", "api.marshal_error", nil, jsonErr.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Write(js)
+}
+
+func getTopChannelsForUserSince(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTimeRange()
+	if c.Err != nil {
+		return
+	}
+
+	c.Params.TeamId = r.URL.Query().Get("team_id")
+
+	// TeamId is an optional parameter
+	if c.Params.TeamId != "" {
+		if !model.IsValidId(c.Params.TeamId) {
+			c.SetInvalidURLParam("team_id")
+			return
+		}
+
+		team, teamErr := c.App.GetTeam(c.Params.TeamId)
+		if teamErr != nil {
+			c.Err = teamErr
+			return
+		}
+
+		if (!team.AllowOpenInvite || team.Type != model.TeamOpen) && !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), team.Id, model.PermissionViewTeam) {
+			c.SetPermissionError(model.PermissionViewTeam)
+			return
+		}
```

Comment:
Is there a reason for a different authz check here compared to [in the `getTopChannelsForTeamSince ` function](https://github.com/mattermost/mattermost-server/pull/19953/files#diff-036c0265e2c7ac93563884c5b029665ae090d47443df092263ce1cd619585246R2113-R2116)?

Reply:
There's no reason other than I missed changing it here as well 😅, thanks for catching that.